File: rpc.go
package unrealircd

import (
	"math/rand"
	"time"
)

// Rpc handles RPC meta operations
type Rpc struct {
	querier Querier
}

// Info gets information on all RPC modules loaded
func (r *Rpc) Info() (interface{}, error) {
	return r.querier.Query("rpc.info", nil, false)
}

// SetIssuer sets the name of the issuer (requires UnrealIRCd 6.0.8+)
func (r *Rpc) SetIssuer(name string) (interface{}, error) {
	return r.querier.Query("rpc.set_issuer", map[string]interface{}{
		"name": name,
	}, false)
}

// AddTimer adds a timer (requires UnrealIRCd 6.1.0+)
func (r *Rpc) AddTimer(timerID string, everyMsec int, method string, params interface{}, id interface{}) (interface{}, error) {
	if id == nil {
		// Generate a random ID above regular query IDs
		rand.Seed(time.Now().UnixNano())
		id = 100000 + rand.Intn(900000)
	}

	request := map[string]interface{}{
		"jsonrpc": "2.0",
		"method":  method,
		"params":  params,
		"id":      id,
	}

	return r.querier.Query("rpc.add_timer", map[string]interface{}{
		"timer_id":   timerID,
		"every_msec": everyMsec,
		"request":    request,
	}, false)
}

// DelTimer deletes a timer (requires UnrealIRCd 6.1.0+)
func (r *Rpc) DelTimer(timerID string) (interface{}, error) {
	return r.querier.Query("rpc.del_timer", map[string]interface{}{
		"timer_id": timerID,
	}, false)
}

File: unrealircd-rpc-cs/UnrealIRCdRPC/Rpc.cs
using System;
using System.Threading.Tasks;
using System.Text.Json;

namespace UnrealIRCdRPC
{
    // Rpc handles RPC meta operations
    public class Rpc
    {
        private readonly IQuerier _querier;

        public Rpc(IQuerier querier)
        {
            _querier = querier;
        }

        // Info gets information on all RPC modules loaded
        public async Task<JsonElement?> InfoAsync()
        {
            return await _querier.QueryAsync("rpc.info", null, false);
        }

        // SetIssuer sets the name of the issuer (requires UnrealIRCd 6.0.8+)
        public async Task<JsonElement?> SetIssuerAsync(string name)
        {
            return await _querier.QueryAsync("rpc.set_issuer", new { name }, false);
        }

        // AddTimer adds a timer (requires UnrealIRCd 6.1.0+)
        public async Task<JsonElement?> AddTimerAsync(string timerId, int everyMsec, string method, object? parameters, object? id = null)
        {
            if (id == null)
            {
                // Generate a random ID above regular query IDs
                var random = new Random();
                id = 100000 + random.Next(900000);
            }

            var request = new
            {
                jsonrpc = "2.0",
                method,
                @params = parameters,
                id
            };

            return await _querier.QueryAsync("rpc.add_timer", new
            {
                timer_id = timerId,
                every_msec = everyMsec,
                request
            }, false);
        }

        // DelTimer deletes a timer (requires UnrealIRCd 6.1.0+)
        public async Task<JsonElement?> DelTimerAsync(string timerId)
        {
            return await _querier.QueryAsync("rpc.del_timer", new { timer_id = timerId }, false);
        }
    }
}